palette/src/relative_contrast.rs

```diff
@@ -0,0 +1,141 @@
+use {from_f64, FromF64};
+
+/// A trait for calculating relative contrast between two colors.
+///
+/// W3C's Web Content Accessibility Guidelines (WCAG) 2.1 suggest a method
+/// to calculate accessible contrast ratios of text and background colors
+/// for those with low vision or color deficiencies, and for contrast of user
+/// interface graphics. The possible range of ratios is from 1:1 to 21:1.
+/// There is a Success Criterion for Contrast (Minimum) and a Success Criterion
+/// for Contrast (Enhanced), SC 1.4.3 and SC 1.4.6 respectively, which are
+/// concerned with text and images of text. SC 1.4.11 is a Success Criterion for
+/// "non-text contrast" such as user interface components and other graphics.
+/// The relative contrast is calculated by `(L1 + 0.05) / (L2 + 0.05)`, where
+/// `L1` is the luminance of the brighter color and `L2` is the luminance of the
+/// darker color both in sRGB linear space. Higher contrast ratio is generally
+/// desireable.
+///
+/// These criteria are recommendations, not hard and fast rules. Manual tests
+/// should be used to check that contrast ratios are compliant in the output
+/// color space. For example, if the input colors are in HSV and the output
+/// colors are in RGB, the contrast should be checked again since the output
+/// color model differs. Conversion of color from HSV to RGB is going to be
+/// lossy in some way so the contrast ratio of the RGB should be re-checked.
```

Comment:
I think this is a good summary, but I feel like I need to clarify a couple of things:

* With "manual tests" I meant actually looking at the result and verifying that it's clearly and comfortably readable, even with low vision or color deficiencies. Nothing beats that, so far.

* The "checking the contrast of the output space" part should probably clarify that it's recommended to basically run `my_background_rgb.has_enhanced_contrast_text(&my_foreground_rgb)` before outputting them, if RGB is the output format (and it usually is).

I would suggest editing it to clarify that and move it up above the more technical part. I think we will have more users than people implementing it. Then I think the example can also be simplified a lot to something that only shows checking for enhanced contrast of a single color pair. Nothing prevents us from having examples for each method too, but I think it's best to have each example as small and on point as possible. It's also more aligned with the rest of the examples if you try to use the types from the crate root, such as `palette::Srgb`. It's the same as `Rgb::<Srgb, _>`, but with less syntax.